src/kbucket/bucket.rs

```diff
@@ -13,34 +13,45 @@ use super::BinaryKey;
 use crate::config::BucketConfig;
 use crate::K_K;
 
+/// Represents a bucket for storing nodes in a Kademlia routing table.
 pub(super) struct Bucket<V> {
     nodes: arrayvec::ArrayVec<Node<V>, K_K>,
     pending_node: Option<Node<V>>,
     bucket_config: BucketConfig,
 }
 
+/// Enum representing the result of inserting a node into a bucket.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum NodeInsertOk<'a, TNode> {
-    Inserted {
-        inserted: &'a TNode,
-    },
+    /// The node was successfully inserted into the bucket.
+    Inserted { inserted: &'a TNode },
+    /// The node was updated within the bucket, and a potential eviction is
+    /// pending.
     Updated {
         updated: &'a TNode,
         pending_eviction: Option<&'a TNode>,
     },
+    /// The insertion of the node is pending, and a potential eviction is
+    /// pending as well.
     Pending {
         pending_insert: &'a TNode,
         pending_eviction: Option<&'a TNode>,
     },
 }
 
+/// Enum representing possible errors when inserting a node into a bucket.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum NodeInsertError<TNode> {
+    /// The node is considered invalid and cannot be inserted.
     Invalid(TNode),
```

Comment:
Having a `MismatchNetwork` error, it means that the `NodeInserError` struct is granular. Because of that, I think it would be more appropriate to rename `Invalid` into `InvalidNonce`